cmd/backup/main.go

```diff
@@ -330,10 +331,21 @@ func (s *script) stopContainers() (func() error, error) {
 // saves it to disk.
 func (s *script) takeBackup() error {
 	s.file = timeutil.Strftime(&s.start, s.file)
-	if err := targz.Compress(s.c.BackupSources, s.file); err != nil {
+  backupCopiedSource := filepath.Join("/tmp", s.c.BackupSources)
+	// copy before compressing guard against a situation where backup folder's content are still growing.
+	if err := copy.Copy(s.c.BackupSources, backupCopiedSource); err != nil {
+		return fmt.Errorf("takeBackup: error creating snapshot on backup folder: %w", err)
+	}
+	if err := targz.Compress(backupCopiedSource, s.file); err != nil {
 		return fmt.Errorf("takeBackup: error compressing backup folder: %w", err)
 	}
 	s.logger.Infof("Created backup of `%s` at `%s`.", s.c.BackupSources, s.file)
+	s.logger.Infof("Removed local artifacts %s.", s.file)
+
+	if err := os.RemoveAll(backupCopiedSource); err != nil {
```

Comment:
Can we `defer` this as we also want to run it in case compressing fails or similar?

Reply:
yeah this is a good point I will fix this